### PR TITLE
[BEAM-14499] Step global, unbounded side input case back to warning

### PR DESCRIPTION
**Pull request by jrmccluskey**
Turn the check for a globally windowed, unbounded side input into a warning rather than an error. This allows the TestStream integration tests to pass while passert streaming functions are developed. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
Run Go Flink ValidatesRunner

**Comment by jrmccluskey** [comment]
Run Go Flink ValidatesRunner

**Comment by jrmccluskey** [comment]
R: @lostluck @damccorm

File: sdks/go/pkg/beam/pardo.go [hunk 1]
```diff
@@ -63,7 +65,8 @@ func TryParDo(s Scope, dofn interface{}, col PCollection, opts ...Option) ([]PCo
 			return nil, fmt.Errorf("main input is global windowed in DoFn %v but side input %v is not, cannot map windows correctly. Consider re-windowing the side input PCollection before use", fn, i)
 		}
 		if (sideWfn.Kind == window.GlobalWindows) && !sideNode.Bounded() {
-			return nil, fmt.Errorf("side input %v is global windowed in DoFn %v but is unbounded, DoFn will block until end of Global Window. Consider windowing your unbounded side input PCollection before use", i, fn)
+			// TODO(BEAM-14501): Replace this warning with an error return when proper streaming test functions have been added.
+			log.Warnf(context.Background(), "side input %v is global windowed in DoFn %v but is unbounded, DoFn will block until end of Global Window. Consider windowing your unbounded side input PCollection before use", i, fn)
```
Comment (damccorm) [inline sdks/go/pkg/beam/pardo.go:69]: If we're planning on upgrading this to an error, I would say that as part of the warning. That saves us the work of needing to do that sort of rollout work in the future.

**Review by damccorm — APPROVED** [review]

File: sdks/go/pkg/beam/pardo.go (hunk 1 above)
Comment (jrmccluskey) [reply sdks/go/pkg/beam/pardo.go]: That's a good note. Added.

**Review by lostluck — COMMENTED** [review]
LGTM

**Pull request merged**